### PR TITLE
Create additional docker image tags for major and minor version

### PR DESCRIPTION
**Pull request by lg2de**
I would like to have tags for major and minor release for more granular updates.

### CODE REVIEW

**Comment by lg2de** [comment]
Please note, that I do not know whether this is working at all. I just tried to adopt existing code...

**Comment by Regenhardt** [comment]
I think this is a good idea, thanks for the PR!  

It looks good to me, but as I'm not well versed in gh actions and docker, I'd prefer another @seriouz look at this.  
Unless there's currently no time, so if it sits a week I'll merge it anyway.

**Comment by seriouz** [comment]
@lg2de @Regenhardt No, that won't work if I think what you want:
If X.Y.Z the X changes the image gets the major tag and if the Y changes the image gets the minor tag. There is no patch tag for Z.
Is that correct? If so - I think it's a very good idea and would have a look at it soon!

**Comment by Regenhardt** [comment]
I thought it tags every created docker image with a cut off part of the version. 

Like the docker image for v1.2.3 will get the tags 1.2.3, 1.2, and 1.  
The image for v1.2.4 will get tagged 1.2.4, 1.2, and 1. 

So if somebody wants to automatically get patches but no features, they can use Bagetter:1.2 and will automatically get the update from 1.2.3 to 1.2.4, but won't get 1.3.0 until they update their tag manually. 

This allows a certain auto update with granular control over if they want to get patches only (:1.2), patches and minor changes (:1), none (:1.2.3), or all (:latest) updates.

**Comment by seriouz** [comment]
Futhermore I think it is possible to tag a Docker image with `:1`. So you will get all minor updates but no major updates:
`v1.1` -> `v1.9` ok
`v1.1` -> `v2.0` not ok

**Comment by lg2de** [comment]
You have described what I was trying to achieve. 
But I don't understand why it shouldn't work that way.

**Comment by seriouz** [comment]
The problem that arises is as follows:
- If you simply always want the latest version, you can roll the `latest` tag (`:latest`). This way you don't care whether `major`, `minor` or `patch` are updated.
- If you are only interested in updates to `minor` and `patch`, but do not want `major` to change, you can roll `:1` as a tag.
- If you are only interested in updates at patch level, you could use the tag `:1.2` to react only to patch updates.
What is the advantage of a `:major` tag? The best I could imagine is that it only reacts to changes to the major version, and that minor and patch changes are **_not_** reacted to. However, this is more of a disadvantage, as most bugs are only fixed in patches and you are more likely to run into problems or security vulnerabilities.

**Comment by lg2de** [comment]
I do not want to have a tag named `major` or `minor`. And, I do not think that my recommended changes are creating such tags.

I guess (and hope): Accepting my changes the next version "1.4.6" will create the following tags: `latest` (overwriting), `1` (now new, later overwriting), `1.4` (now new, later overwriting), and `1.4.6` (new, only once).

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -13,9 +13,15 @@ jobs:
         run: |
           export GITHUB_REF_NAME="${{ github.ref_name }}"
           export RELEASE_VERSION="${GITHUB_REF_NAME#v}"
+          export RELEASE_MINOR_VERSION="${RELEASE_VERSION%.*}"
+          export RELEASE_MAJOR_VERSION="${RELEASE_MINOR_VERSION%.*}"
```
Comment (seriouz) [inline .github/workflows/release.yml:17]: To be prepared for other variants of the semver2 tags i would slightly modifiy the script to pull out major and minor:

```suggestion
          export RELEASE_MAJOR_VERSION="${RELEASE_VERSION%%.*}"
          export remaining_version="${RELEASE_VERSION#*.}"
          export RELEASE_MINOR_VERSION="${RELEASE_MAJOR_VERSION}.${remaining_version%%.*}"

```

**Review by seriouz — CHANGES_REQUESTED** [review]
I was stumped, but now I know what you meant, sorry.
So far it looks very good, I just have a small change suggestion that better covers other possible SemVer2 variants.

File: .github/workflows/release.yml (hunk 1 above)
Comment (lg2de) [reply .github/workflows/release.yml:17]: For which kind of tag do we get a difference?
I'm not that familiar with the substitution syntax...

**Pull request merged**